satpy/readers/fci_l1c_nc.py

```diff
@@ -91,6 +93,11 @@
 
 import logging
 
+# Importing hdf5plugin makes decompression plugins available and work automatically when reading data
+try:
+    import hdf5plugin  # noqa: F401
+except ImportError:
+    pass
```

Comment:
Could you replace this with `hdf5plugin = None` and then in the `__init__` of the file reader detect that it isn't there and raise an error if the files are compressed? Or is the existing error message pretty clear?

Reply:
I've yet to find how to determine whether the data are compressed or not. But when I do, I'll do as you suggest. The full trace when trying to read the data without `hdf5plugin` is:
```python
Traceback (most recent call last):
  File "/home/lahtinep/bin/test_fci.py", line 84, in <module>
    main()
  File "/home/lahtinep/bin/test_fci.py", line 71, in main
    lcl.save_datasets(base_dir='/tmp/', tiled=True, blockxsize=512, blockysize=512, driver='COG', overviews=[])
  File "/home/lahtinep/Software/pytroll/pytroll_packages/satpy/satpy/scene.py", line 1144, in save_datasets
    return writer.save_datasets(dataarrays, compute=compute, **save_kwargs)
  File "/home/lahtinep/Software/pytroll/pytroll_packages/satpy/satpy/writers/__init__.py", line 691, in save_datasets
    return compute_writer_results([results])
  File "/home/lahtinep/Software/pytroll/pytroll_packages/satpy/satpy/writers/__init__.py", line 536, in compute_writer_results
    da.compute(delayeds)
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/base.py", line 571, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/threaded.py", line 79, in get
    results = get_async(
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/local.py", line 507, in get_async
    raise_exception(exc, tb)
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/local.py", line 315, in reraise
    raise exc
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/local.py", line 220, in execute_task
    result = _execute_task(task, data)
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/lahtinep/mambaforge/envs/pytroll/lib/python3.9/site-packages/dask/array/core.py", line 110, in getter
    c = a[b]
  File "src/netCDF4/_netCDF4.pyx", line 4406, in netCDF4._netCDF4.Variable.__getitem__
  File "src/netCDF4/_netCDF4.pyx", line 5350, in netCDF4._netCDF4.Variable._get
  File "src/netCDF4/_netCDF4.pyx", line 1927, in netCDF4._netCDF4._ensure_nc_success
RuntimeError: NetCDF: Filter error: unimplemented filter encountered
```
The processing doesn't stop there, and empty images were saved.

Reply:
As discussed in Slack we could use the `special_compression` key from the `filename_info`, which is empty if uncompressed and `JLS` if compressed.